aspects/aspects.go

```diff
@@ -331,9 +345,51 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 	return notFoundErrorFrom(a, request, "no matching write rule")
 }
 
+// namespaceResult creates a nested namespace around the result that corresponds
+// to the unmatched entry parts. Unmatched placeholders are filled in using maps
+// of all the matching values in the databag.
+func namespaceResult(res interface{}, suffixParts []string) (interface{}, error) {
+	if len(suffixParts) == 0 {
+		return res, nil
+	}
+
+	// check if the part is an unmatched placeholder which should have been filled
+	// by the databag with all possible values
+	part := suffixParts[0]
+	if part[0] == '{' && part[len(part)-1] == '}' {
+		values, ok := res.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("internal error: expected databag to return map for unmatched placeholder")
+		}
+
+		level := make(map[string]interface{}, len(values))
+		for k, v := range values {
+			nested, err := namespaceResult(v, suffixParts[1:])
+			if err != nil {
+				return nil, err
+			}
+
+			level[k] = nested
+		}
+
+		return level, nil
+	}
+
+	nested, err := namespaceResult(res, suffixParts[1:])
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{part: nested}, nil
+}
+
 // Get returns the aspect value identified by the request. If either the named
 // aspect or the corresponding value can't be found, a NotFoundError is returned.
 func (a *Aspect) Get(databag DataBag, request string, value *interface{}) error {
```

Comment:
unrelaed: given we do no casting on behalf of the caller atm, it seems having Get returning value instead would make sense, same for Transaction.Get